Sources/Apodini/Semantic Model Builder/Relationship/RelationshipCandidate.swift

```diff
@@ -0,0 +1,201 @@
+//
+// Created by Andreas Bauer on 23.01.21.
+//
+
+enum RelationshipType: Hashable, CustomStringConvertible, CustomDebugStringConvertible {
+    /// All relationships from the destination are inherited (including the self relationship).
+    /// This requires that the `RelationshipSourceCandidate` contains resolvers
+    /// for all path parameters of the relationship destination.
+    case inheritance
+    /// A reference describes a special case of Relationship where properties
+    /// of the returned data is REPLACED by the relationship (e.g. a link in the REST exporter).
+    /// This type requires that the `RelationshipSourceCandidate` contains resolvers
+    /// for all path parameters of the relationship destination.
+    case reference(name: String)
+    /// A link is any kind of Relationship. It doesn't enforce any path parameters to be resolved
+    /// (although it can be resolved if wanted).
+    case link(name: String)
+
+    var description: String {
+        switch self {
+        case .inheritance:
+            return "inheritance"
+        case .reference:
+            return "reference"
+        case .link:
+            return "link"
+        }
+    }
+
+    var debugDescription: String {
+        switch self {
+        case .inheritance:
+            return "inheritance"
+        case let .reference(name):
+            return "reference(name: \"\(name)\")"
```

Comment:
```suggestion
            return #"reference(name: "\(name)")"#
```
They are _Extended String Delimiters_, if you didn't already know 🙂

Reply:
Your example does not work as intended, as you seemingly also need to reescape the string Interpolation.
`#"reference(name: "\#(name)")"#`. Not sure what reads better.

Reply:
Resolved via 6c2b9bf